scripts/fetch-asyncapi-example.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+const request = require('request');
+const fs = require('fs');
+const unzipper = require('unzipper');
+const path = require('path');
+const parser = require('@asyncapi/parser');
+
+const SPEC_EXAMPLES_ZIP_URL = 'https://github.com/asyncapi/spec/archive/refs/heads/master.zip';
+const EXAMPLE_DIRECTORY = path.join(__dirname, '../assets/examples');
+const TEMP_ZIP_NAME = 'spec-examples.zip';
+
+const fetchAsyncAPIExamplesFromExternalURL = () => {
+  return new Promise((resolve) => {
+    request(SPEC_EXAMPLES_ZIP_URL)
+      .pipe(fs.createWriteStream(TEMP_ZIP_NAME))
+      .on('close', () => {
+        console.log('Fetched ZIP file');
+        resolve();
+      });
+  });
+};
+
+const unzipAsyncAPIExamples = async () => {
+  return new Promise((resolve, reject) => {
+    if (!fs.existsSync(EXAMPLE_DIRECTORY)) {
+      fs.mkdirSync(EXAMPLE_DIRECTORY);
+    }
+
+    fs.createReadStream(TEMP_ZIP_NAME)
+      .pipe(unzipper.Parse())
+      .on('entry', async (entry) => {
+        const fileName = entry.path;
+        if (fileName.includes('examples/') && fileName.includes('.yml') && entry.type === 'File') {
+          const fileContent = await entry.buffer();
+          const fileNameWithExtension = fileName.split('examples/')[1];
+          fs.writeFileSync(path.join(EXAMPLE_DIRECTORY, fileNameWithExtension), fileContent.toString());
+        } else {
+          entry.autodrain();
+        }
+      }).on('close', () => {
+        console.log('Unzipped all examples from zip');
+        resolve();
+      }).on('error', () => {
+        reject();
+      });
+  });
+};
+
+const buildCLIListFromExamples = async () => {
+  const files = fs.readdirSync(EXAMPLE_DIRECTORY);
+  const examples = files.filter(file => file.includes('.yml')).sort();
+
+  const listAllProtocolsForFile = (parsedAsyncAPI) => {
+    if (!parsedAsyncAPI.hasServers()) {return '';}
+    const servers = parsedAsyncAPI.servers();
+    return Object.keys(servers).map(server => servers[String(server)].protocol()).join(',');
+  };
+
+  const buildExampleList = examples.map(async example => {
+    const examplePath = path.join(EXAMPLE_DIRECTORY, example);
+    const exampleContent = fs.readFileSync(examplePath, { encoding: 'utf-8'});
+    
+    try {
+      const parsedSpec = await parser.parse(exampleContent);
+      const title = parsedSpec.info().title();
+      const protocols = listAllProtocolsForFile(parsedSpec);
+      return {
+        name: protocols ? `${title} - (protocols: ${protocols})` : title,
+        value: example
+      };
+    } catch (error) {
+      // Failed for somereason to parse this spec file, ignore for now
+    }
```

Comment:
You should at least log that error

```suggestion
    } catch (error) {
      // Failed for somereason to parse this spec file, ignore for now
      console.error(error);
    }
```

Reply:
Cool thanks @magicmatatjahu